mlflow/gateway/providers/base.py

```diff
@@ -16,26 +16,44 @@ class BaseProvider(ABC):
     SUPPORTED_ROUTE_TYPES: Tuple[str, ...]
 
     def __init__(self, config: RouteConfig):
+        if not hasattr(self, "NAME"):
+            raise TypeError(f"'NAME' is not defined for {self.__class__.__name__}")
+
         self.config = config
 
     async def chat_stream(
         self, payload: chat.RequestPayload
     ) -> AsyncIterable[chat.StreamResponsePayload]:
-        raise NotImplementedError
+        raise HTTPException(
+            status_code=404,
+            detail=f"The chat streaming route is not available for {self.NAME} models.",
+        )
 
     async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
-        raise NotImplementedError
+        raise HTTPException(
+            status_code=404,
+            detail=f"The chat route is not available for {self.NAME} models.",
+        )
 
     async def completions_stream(
         self, payload: completions.RequestPayload
     ) -> AsyncIterable[completions.StreamResponsePayload]:
-        raise NotImplementedError
+        raise HTTPException(
+            status_code=404,
+            detail=f"The completions streaming route is not available for {self.NAME} models.",
```

Comment:
```suggestion
            detail=f"The completions streaming route is not implemented for {self.NAME} models.",
```

Is `implemented` more accurate?

Reply:
i agree